io/disk/softwareraid.py

```diff
@@ -45,69 +45,93 @@ def setUp(self):
         """
 
         smm = SoftwareManager()
-        if smm.check_installed("mdadm") is False:
-            print("Mdadm is not installed")
+        if not smm.check_installed("mdadm"):
+            self.skip("Mdadm must be installed before continuing the test")
             if SoftwareManager().install("mdadm") is False:
                 self.skip("Unable to install mdadm")
         cmd = "mdadm -V"
-        if process.system(cmd, ignore_status=True, shell=True) != 0:
-            self.fail("Unable to get the mdadm Version")
-        self.loop = ''
-        for _ in range(0, 4):
-            cmd = "losetup --find"
-            varstr = process.system_output(cmd, ignore_status=True, shell=True)
-            var = int(re.match(r'(/dev/loop)(\d+)', varstr).group(2))
-            cmd = "dd if=/dev/zero of=/file%d bs=1k count=100" % var
-            process.system(cmd, ignore_status=True, shell=True)
-            cmd = "losetup /dev/loop%d /file%d" % (var, var)
-            process.system(cmd, ignore_status=True, shell=True)
-            self.loop += "/dev/loop%d " % var
-        self.disk = self.params.get('disk', default=self.loop).strip(" ")
+        self.check_pass(cmd, "Unable to get mdadm version")
+        self.disk = self.params.get('disk', default='').strip(" ")
+        self.disk = self.disk.split(" ")
```

Comment:
why not `self.disk = self.params.get('disk', default='').strip(" ").split(" ")` ?